### ISSUE TITLE
[HIGH SEVERITY] Laptop internal screen **doesn't turn on** after running apt upgrade

### ISSUE THREAD

**Issue by lpaderiAltravia** [pop-os/pop#1722]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
 

**Issue/Bug Description:**
My HP Pavillon's internal screen doesnt turn on any more after running a system upgrade and reboot. The issue was also reported by [others](https://www.reddit.com/r/pop_os/comments/nsdg4c/build_in_screen_black_after_updating_os/) with apparently similar setups.

I also saw two nvidia driver upgrades today..

External screens work just fine. I am not using and external monitor and mirroring the internal display, otherwise my laptop would be unusable

**Comment by bflanagin** [comment]
Beyond trying an external screen on your laptop have you tried anything else? 
What happens when you toggle the display button (usually fn+F7)?

**Comment by lpaderiAltravia** [comment]
This is what I have tried

- Turn up the brightness, both with hw keys and  the sliders
- Switch graphic card. Now is set to descrete
- The procedure in your website, about re-installing the driver

Then I did:
- A system refresh from a USB key

This one worked, but as soon as I installed the new upgrades the issue re appeared.

**Comment by lpaderiAltravia** [comment]
# UPDATE

For those of you reading: the source of the problem seems to be (is)  the new kernel, `5.11.0-7614`

But you can still install an older version.

**Comment by lpaderiAltravia** [comment]
The issue was closed by mistake, as this is not a solution but a workaround. I hope the team can have a look and fix it ASAP

**Comment by prashantpro** [comment]
This is a serious issue. Few things I myself ran into:
1) POP OS 21.04 Beta - boots into blank screen (can't install)
2) POP OS 20.10 installed version, try to upgrade - Laptop screen is no longer detected
 - Used my HDMI cable connected to external monitor (which works) but using xrandr or other options suggests that the internal screen is disconnected.
 - Trying to switch to any of the TTYs don't work either. Still blank screen.
3) Few months back I had tried Arch Linux with 5.12 kernel and too landed me on a blank screen

Seems all 3 issues above could be related to kernel itself.
While I can avoid going for the Beta 21.04, the updates are very usual flow for anyone and ideally should not break the system.

Impacted laptops are probably the Optimus Laptops (I use Optimus - HP Omen laptop with hybrid graphics).

**Comment by halathamneh** [comment]
I'm using Lenovo Legion Y520 (with Nvidia GTX 1050 Ti) and facing the same problem, I had to downgrade to `5.8.0-7642` in order to get it back working.
Downgrading to `5.11.0-7612` didn't work as well, the same issue, internal screen is blank.

**Comment by prashantpro** [comment]
@halathamneh I had downgraded to 5.10 and that worked as well, in case you wish to try.
It's a problem with kernel beyond a certain version number of 5.11.x.

I see few options here:

- Pop OS 21.04 should stick to a stable kernel similar to Ubuntu 21.04 i.e. 5.11.0-18-generic
- Wait for Kernel level fixes (which I'm not sure is coming anytime soon).

For now, I have switched to Ubuntu 21.04 with System76 driver and Pop GTK theme. 
https://github.com/pop-os/gtk-theme

Planning to stay with Ubuntu for now, till a fix is provided within Pop OS. (really hoping for one soon).

**Comment by lpaderiAltravia** [comment]
If you need to fix black screen after 21.04 upgrade, create a new file in `/boot/efi/loader/entries` called `whatever.conf`

```
title Pop!_OS
linux /EFI/Pop_OS-your-system-uuid/vmlinuz-5.11.0-7612-generic
initrd /EFI/Pop_OS-your-system-uuid/initrd.img-5.11.0-7612-generic
options root=UUID=enter_your_system_uuid ro quiet loglevel=0 systemd.show_status=false splash
```

Your system UUID can be checked with
```
sudo ls /boot/efi/EFI | grep Pop
```
Pop_OS-uuid_here

**Comment by N0tJust1n** [comment]
I downgraded to kernel 5.10.53 and still observed this issue. Going to try 5.8 tonight. Is this on the radar of the kernel maintainers? I'm not sure how they track issues like this?

**Comment by N0tJust1n** [comment]
5.8 results in all programs not being able to see my GPU (1660ti) and 5.10 still has the issue for me. What else can I do?

**Comment by lpaderiAltravia** [comment]
I think it got lost among many. Hopefully these comments will bump it up

**Comment by jacobgkau** [comment]
> Pop OS 21.04 should stick to a stable kernel similar to Ubuntu 21.04 i.e. 5.11.0-18-generic

This request doesn't quite make sense; the Pop!_OS kernel is already based on the Ubuntu kernel. You can see the changes made to better support System76 hardware here: https://github.com/pop-os/linux/commits/master (Anything below "System76 Linux" is from Ubuntu's own packaging.)

> Few months back I had tried Arch Linux with 5.12 kernel and too landed me on a blank screen

This sounds like it's an upstream kernel issue, and seems contrary to the issue _not_ occurring on Ubuntu 21.04 with kernel 5.11. Does the issue occur with mainline kernel 5.12 or 5.13 on Ubuntu? Does the issue still not occur on Ubuntu 21.04 with its default kernel?

Has anyone tried this Optimus workaround for GDM? https://wiki.archlinux.org/title/NVIDIA_Optimus#GDM 

---

System76 customers can reach out to [support](https://support.system76.com/) for technical assistance. For non-System76 hardware, you can seek community support on [Reddit](https://reddit.com/r/pop_os/) or [Mattermost](https://chat.pop-os.org).

**Comment by lpaderiAltravia** [comment]
> > Pop OS 21.04 should stick to a stable kernel similar to Ubuntu 21.04 i.e. 5.11.0-18-generic
> 
> This request doesn't quite make sense; the Pop!_OS kernel is already based on the Ubuntu kernel. You can see the changes made to better support System76 hardware here: https://github.com/pop-os/linux/commits/master (Anything below "System76 Linux" is from Ubuntu's own packaging.)
> 
> > Few months back I had tried Arch Linux with 5.12 kernel and too landed me on a blank screen
> 
> This sounds like it's an upstream kernel issue, and seems contrary to the issue _not_ occurring on Ubuntu 21.04 with kernel 5.11. Does the issue occur with mainline kernel 5.12 or 5.13 on Ubuntu? Does the issue still not occur on Ubuntu 21.04 with its default kernel?
> 
> Has anyone tried this Optimus workaround for GDM? https://wiki.archlinux.org/title/NVIDIA_Optimus#GDM
> 
> System76 customers can reach out to [support](https://support.system76.com/) for technical assistance. For non-System76 hardware, you can seek community support on [Reddit](https://reddit.com/r/pop_os/) or [Mattermost](https://chat.pop-os.org).

Hi jakob, thanks for the reply.

I will try the workaround as soon as possible; about ubuntu 21.04, unfortunately I need my laptop for work and I don't really want to play around with it too much, as it would probably mean to do some hacks to change the kernel (the live ISO is working no problem, as it is using an older kernel) and I am not that skilled. 

However, you can understand that this is a serious issue that makes the laptop unusable

**Comment by jacobgkau** [comment]
> I will try the workaround as soon as possible; about ubuntu 21.04, unfortunately I need my laptop for work and I don't really want to play around with it too much,

> However, you can understand that this is a serious issue that makes the laptop unusable

Because you are not using System76 hardware, we don't have an affected machine to test with, so we are reliant on your testing to determine the root cause and the effectiveness of any fixes. Testing on upstream Ubuntu is very important because if the issue also occurs in Ubuntu, this would need to be fixed either in Ubuntu's kernel or in the upstream Linux kernel, not in Pop!_OS. Please let us know if/when you're able to confirm that.

If the issue does occur in Ubuntu, then the appropriate place to report this would be Ubuntu's bug tracker here: https://bugs.launchpad.net/ubuntu/+source/linux

**Comment by lpaderiAltravia** [comment]
> If the issue does occur in Ubuntu, then the appropriate place to report this would be Ubuntu's bug tracker here: https://bugs.launchpad.net/ubuntu/+source/linux

https://bugs.launchpad.net/ubuntu/+source/linux-hwe-5.8/+bug/1934647

I was totaly unaware of the existence of this tool, however, a quick search reveals a similar problem to the one we are talking about.

If anyone has more information or is willing to explain me how to debug issues like these, I will provide more logs. As for now, this thread can probably be closed.
Thank you

**Comment by prashantpro** [comment]
@jacobgkau it's surely a kernel issue and not specific to distros like Pop OS or Ubuntu.
My comment earlier was "Seems all 3 issues above could be related to kernel itself." This seems to be confirmed by how this discussion has transitioned.

> Pop OS 21.04 should stick to a stable kernel similar to Ubuntu 21.04 i.e. 5.11.0-18-generic

When I had initially shared my comment on this issue it was when things were working in Ubuntu 21.04 but not in Pop OS 21.04.
Pop OS 21.04 was on a kernel higher than Ubuntu's - "5.11.0-7614" (That minor version bump is known to have some issues).
My hope was that maybe the Ubuntu team was aware of the issue and hence didn't roll an update out (on the kernel). But guess not, as Ubuntu has recently also upgraded the kernel.

In short - Ubuntu and Pop OS has adopted the recent kernel which does cause issues with Optimus Laptops but given it's a kernel thingy, this issue doesn't belong here. I will certainly try a newer kernel again.
For now, happily using Pop OS 21.04 with this downgraded kernel
"5.10.47-051047-generic"

Just shared to clarify.

**Comment by N0tJust1n** [comment]
Another workaround some might try when they search the issue:

I did a refresh install from recovery last night and when I rebooted into the desktop BEFORE doing updates on pop_shop I put a hold on all *nvidia* packages and the few kernel packages that were showing as upgradable in synaptic. This leaves me firmly planted on nvidia-driver-465 and kernel 5.11.0-7620. Maybe the issue is an incompatibility between certain nvidia drivers and kernels but this combination seems to work for me were 5.10 or 5.8 did not.
